### PR TITLE
resource_group: supports burstable for resource group

### PR DESCRIPTION
**Pull request by nolouch**
Signed-off-by: nolouch <nolouch@gmail.com>

### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/40380

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [x] Manual test (add detailed scripts or steps below)
Details in: https://asciinema.org/a/552982
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by BornChanger — COMMENTED** [review]
Expected result of `TestKeywordConsistent` needs to be fixed.

File: ddl/ddl_api.go [hunk 1]
```diff
@@ -3046,6 +3046,13 @@ func SetDirectResourceGroupUnit(resourceGroupSettings *model.ResourceGroupSettin
 		resourceGroupSettings.IOReadBandwidth = stringVal
 	case ast.ResourceUnitIOWriteBandwidth:
 		resourceGroupSettings.IOWriteBandwidth = stringVal
+	case ast.ResourceBurstableOpiton:
+		limit := int64(0)
+		if boolValue {
+			// negative means no limit within burst setting.
+			limit = -1
```
Comment (BornChanger) [inline ddl/ddl_api.go:3053]: What's the different between -1 and 0?

Comment (nolouch) [reply ddl/ddl_api.go:3056]: I add some comments, it's not by convention:)

**Comment by glorv** [comment]
need rebase after https://github.com/pingcap/tidb/pull/40884 is merged.

**Comment by glorv** [comment]
@nolouch Please resolve the conflicts

File: ddl/ddl_api.go [hunk 2]
```diff
@@ -3044,6 +3044,16 @@ func SetDirectResourceGroupUnit(resourceGroupSettings *model.ResourceGroupSettin
 		resourceGroupSettings.IOReadBandwidth = stringVal
 	case ast.ResourceUnitIOWriteBandwidth:
 		resourceGroupSettings.IOWriteBandwidth = stringVal
+	case ast.ResourceBurstableOpiton:
+		// Some about BurstLimit(b):
+		//   - If b == 0, that means the limiter is unlimited capacity. default use in resource controller (burst with a rate within a unlimited capacity).
+		//   - If b < 0, that means the limiter is unlimited capacity and filerate(r) is ignored, can be seen as r == Inf (burst with a inf rate within a unlimited capacity).
```
Comment (BornChanger) [inline ddl/ddl_api.go:3050]:
```suggestion
		//   - If b < 0, that means the limiter is unlimited capacity and fill rate(r) is ignored, can be seen as r == Inf (burst with a inf rate within a unlimited capacity).
```

**Review by glorv — APPROVED** [review]

**Pull request merged**